### PR TITLE
apidump: Fix whitespace between names & types

### PR DESCRIPTION
**Pull request by charles-lunarg**
setw() will reset back to 0 in many cases, one of which includes << This meant that call to `stream << setw();` does not apply the proper whitespace required, leading to not adding any space between the name and type. `setw() << "";` forces the correct width to be printed.

### CODE REVIEW

**Review by christophe-lunarg — APPROVED** [review]
Looks good to me.

File: layersvt/api_dump.h [hunk 1]
```diff
@@ -519,18 +524,20 @@ class ApiDumpSettings {
 
     std::ostream &formatNameType(std::ostream &stream, int indents, const char *name, const char *type) const {
         stream << indentation(indents) << name << ": ";
-
         if (use_spaces)
-            stream << std::setfill(' ') << std::setw(name_size - (int)strlen(name) - 2);
+            stream << std::setw(name_size - (int)strlen(name) - 2) << "";
```
Comment (lunarpapillo) [inline layersvt/api_dump.h:528]: Your commit message explains this construction, but it's likely a future engineer won't review the commit messages to figure out why this "needless" construction is done.  Add a comment here explaining why the unusual construction is used.

File: layersvt/api_dump.h [hunk 2]
```diff
@@ -441,6 +441,11 @@ class ApiDumpSettings {
             stream() << "[\n";
         }
 
+        if (use_spaces)
+            stream() << std::setfill(' ');
+        else
+            stream() << std::setfill('\t');
+
```
Comment (lunarpapillo) [inline layersvt/api_dump.h:448]: This seems unrelated to the commit comment.  It's also clearly a behavioral change; its correctness is unclear from context.  Please explain.

**Review by lunarpapillo — CHANGES_REQUESTED** [review]

File: layersvt/api_dump.h (hunk 2 above)
Comment (charles-lunarg) [reply layersvt/api_dump.h:448]: Check the api dump refactor PR - this commit is in that and I address that issue there (with a comment)

File: layersvt/api_dump.h (hunk 1 above)
Comment (charles-lunarg) [reply layersvt/api_dump.h:528]: Also added comment about that in the other PR.

**Comment by charles-lunarg** [comment]
Closing this because #1771 has this commit and fixes some of the mistakes with it.
